src/types/internal.rs

```diff
@@ -113,3 +113,6 @@ pub(crate) trait Callback<T: Clone + Copy + Sized>: Sized {
         }
     }
 }
+
+/// This type marks the `Arguments` trait as sealed.
+pub trait ArgumentsInternal {}
```

Comment:
My preference is a `trait Sealed` in a `mod private`. This is a common pattern and it makes the docs more clear since the user will still see the super trait.

```rust
pub trait Arguments<'a>: ArgumentsInternal { }

// vs

pub trait Arguments<'a>: private::Sealed { }
```